### ISSUE TITLE
TF Lite (CMake): Does not compile on Windows

### ISSUE THREAD

**Issue by stewartmiles** [tensorflow/tensorflow#55745]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

2.9.0

### Custom Code

No

### OS Platform and Distribution

Windows (MSVC 2019)

### Mobile device

N/A

### Python version

N/A

### Bazel version

N/A

### GCC/Compiler version

N/A

### CUDA/cuDNN version

N/A

### GPU model and memory

N/A

### Current Behaviour?

```shell
TF Lite fails to compile on Windows with MSVC 2019 due to `M_PI` being used by `tensorflow/core/lib/random/random_distributions_utils.h` without defining `_USE_MATH_DEFINES` (see https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-160).
```

### Standalone code to reproduce the issue

```shell
Attempt to compile TF Lite on Windows with CMake and it will fail to build with the following error:

random_ops.cc
build\tensorflow\tensorflow/core/lib/random/random_distributions_utils.h(78,27): error C2065: 'M_PI': undeclared identifier
 [build\_deps\tensorflow-build\tensorflow-lite.vcxproj]
\build\tensorflow\tensorflow/core/lib/random
/random_distributions_utils.h(78,15): error C2737: 'v1': const object must be initialized [build\_deps\tensorflow-build\tensorflow-lite.vcxproj]
```
```


### Relevant log output

_No response_</details>

**Comment by stewartmiles** [comment]
After fixing that issue, there is another problem:

```
external_delegate.cc
src\tensorflow\tensorflow\lite\delegates\external\external_delegate.cc(158,11): error C7555: use of designated initializers requires at least '/std:c++20' [src\tensorflow\tensorflow\lite\cmakebuild\tensorflow-lite.vcxproj]
```

has TF _really_ moved to C++20? It doesn't seem great for compatibility.

**Comment by stewartmiles** [comment]
Looks like some C99 (which, AFAIK, didn't appear until C++20) has crept into TF Lite.
https://github.com/tensorflow/tensorflow/blob/0cdb708bf66dde960e4ff9d6e8bf304d7d2b6d7c/tensorflow/lite/delegates/external/external_delegate.cc#L157

**Comment by mohantym** [comment]
Ok @stewartmiles ! Thanks for reporting the issue. This Issue will be closed once PR #55746 is merged. Thanks!
